### DISCUSSION TITLE
What should we plug in?

### DISCUSSION THREAD

**Discussion by loleg** [dribdat/design#17] [Polls]
There are [various ideas in our issues](https://github.com/dribdat/dribdat/labels/integration) of prototyping and planning tools to connect into Dribdat. You can already easily embed an [Etherpad](https://docs.etherpad.org/demo.html)* for quick text collaboration, for example - but we can make this even more plug and play - and endorse open and accessible tools that help the teams.

_(* note: if your Etherpad server shows errors in an iframe, ask the admin to [update it](https://github.com/ether/etherpad-lite/issues/6618))_

Leave comment please with any other suggestions :point_down:

**Comment by derrickoswald** [comment]
I have not evaluated any of the options I voted for (OpenRefine, Superset), but in my limited experience a huge percentage of the effort in a hackathon is getting data into a form that's usable by the "solution" being worked on.

This usually involves some sequence of [data wrangling](https://en.wikipedia.org/wiki/Data_wrangling), probably best captured as a Jupyter or Zeppelin script, for example, that can be repeated by people arriving late to the party.

In my mind, the ideal scenario has: the hackathon project page has a link that opens a browser page to fetch the data it uses from open data sources and processes it down to the format used by the app/web-page that is the end-result/demo of the "hack day".

**Reply by florihas** [reply]
+1

**Comment by loleg** [comment]
There are two ways to do what we are talking about in Dribdat:

1. Create a project that describes or even pre-configures the embedding of a tool in a **Templates** area. This only works for new projects/prototypes - not an issue if you're grouping challenges using the **Category**.
2. By customizing `stages.yaml` you can assign links to project tools a stage, and have it recommended to the users appropriately as they **Post** to their project. 

After thinking about your comments for a while and chatting with folks in the open data community ( :wave: hey Paul !), it seems to me that an even more promising way to develop this would be through continuous crowdsourcing. In other words, instead of having to decide individual project tools to natively integrate with, we would have some opinionated defaults and let organizers fill others into the database, along with the details of how to generate the URL or other instructions.

Further on we could expand this alternatives list with a database that we crowdsource (like the sites below), or plug in a comparison service (does [AlternativeTo](https://alternativeto.net/about/) or [LibHunt](https://www.libhunt.com/) have an API?), that does this even more widely and intelligently. Here are places to discover cool project tools:

- [Selfh.st](https://selfh.st/)
- [Ethical.net](https://ethical.net/)
- [EU alternatives](https://european-alternatives.eu/categories)
- [awesome-foss-alternatives](https://github.com/sfermigier/awesome-foss-alternatives)
- [awesome-self-hosted](https://github.com/awesome-selfhosted/awesome-selfhosted)

**Comment by loleg** [comment]
I propose a simple initial solution. Give the user a little bit of advice about where to find a tool and how to plug it in. Do it in the place where they are most likely to need it: the project editor. For the [next release](https://github.com/dribdat/dribdat/pull/466), I'm testing a new :rocket: **Tools** button that pops up a dialog like this:

![](https://github.com/user-attachments/assets/d097af3b-c2e1-49e4-9905-26b3cfb32b1c)

Adding the link and tapping the button makes it your current Presentation. Note that this link versioned along with the text, so we could give the users the chance to go back to a previous version of an upload, or a tool they shared at an earlier stage with their team.

I would love to include a huge list of tools from [Planka](https://planka.app/) to [Penpot](https://www.wearecube.ch/blog/konkurrenz_fuer_figma_ein_kurzer_blick_auf_Penpot_2_0/) in there, with strong preference to apps that can be plugged in with one or two clicks (while most cloud apps require you to set up an account, then giving access one-by-one to your team members). For now these are just a couple of very opinionated suggestions - with an invitation to join this thread, and share your ideas of how to expand on this!
